packages/core/test/ifthenelse_test.js

```diff
@@ -0,0 +1,424 @@
+          then: {
+            $ref: "#/definitions/unitedkingdom",
+          },
+        },
+      ],
+    };
+
+    it("should render correctly when condition is true when then contains a reference", () => {
+      const formData = {
+        country: "United Kingdom",
+      };
+
+      const { node } = createFormComponent({
+        schema: schemaWithAllOfRef,
+        formData,
+      });
+
+      expect(node.querySelector("input[label=postcode]")).not.eql(null);
+    });
+  });
+
+  it("handles nested if then else", () => {
+    const schemaWithNested = {
+      $schema: "http://json-schema.org/draft-07/schema#",
+      type: "object",
+      properties: {
+        country: {
+          enum: ["USA"],
+        },
+      },
+      required: ["country"],
+      if: {
+        properties: {
+          country: {
+            const: "USA",
+          },
+        },
+        required: ["country"],
+      },
+      then: {
+        properties: {
+          state: {
+            type: "string",
+            anyOf: [
+              {
+                const: "California",
+              },
+              {
+                const: "New York",
+              },
+            ],
+          },
+        },
+        required: ["state"],
+        if: {
+          properties: {
+            state: {
+              const: "New York",
+            },
+          },
+          required: ["state"],
+        },
+        then: {
+          properties: {
+            city: {
+              type: "string",
+              anyOf: [
+                {
+                  const: "New York City",
+                },
+                {
+                  const: "Buffalo",
+                },
+                {
+                  const: "Rochester",
+                },
+              ],
+            },
+          },
+        },
+        else: {
+          if: {
+            properties: {
+              state: {
+                const: "California",
+              },
+            },
+            required: ["state"],
+          },
+          then: {
+            properties: {
+              city: {
+                type: "string",
+                anyOf: [
+                  {
+                    const: "Los Angeles",
+                  },
+                  {
+                    const: "San Diego",
+                  },
+                  {
+                    const: "San Jose",
+                  },
+                ],
+              },
+            },
+          },
+        },
+      },
+    };
+
+    const formData = {
+      country: "USA",
+      state: "California",
+    };
+    const { node } = createFormComponent({
+      schema: schemaWithNested,
+      formData,
+    });
+    expect(node.querySelector("select[id=root_country]")).not.eql(null);
+    expect(node.querySelector("select[id=root_state]")).not.eql(null);
+    expect(node.querySelector("select[id=root_city]")).not.eql(null);
+  });
+
+  it("handles additionalProperties with if then else", () => {
```

Comment:
Test we fixed the issue in this comment: https://github.com/rjsf-team/react-jsonschema-form/pull/1666#discussion_r640334787